modules/porous_flow/test/tests/sinks/s11.i

```diff
@@ -0,0 +1,135 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 3
+  nx = 5
+  ny = 5
+  nz = 5
+  xmin = 0
+  xmax = 10
+  ymin = 0
+  ymax = 10
+  zmin = 0
+  zmax = 10
+[]
+
+[GlobalParams]
+  PorousFlowDictator = dictator
+[]
+
+[UserObjects]
+  [./dictator]
+    type = PorousFlowDictator
+    porous_flow_vars = 'pp temp'
+    number_fluid_phases = 1
+    number_fluid_components = 1
+  [../]
+  [./pc]
+    type = PorousFlowCapillaryPressureVG
+    m = 0.5
+    alpha = 1
+  [../]
+[]
+
+[Variables]
+  [./pp]
+    initial_condition = 1
+  [../]
+  [./temp]
+    initial_condition = 2
+  [../]
+[]
+
+[Kernels]
+  [./mass0]
+    type = PorousFlowMassTimeDerivative
+    fluid_component = 0
+    variable = pp
+  [../]
+
+  [./heat_conduction]
+    type = TimeDerivative
+    variable = temp
+  [../]
+[]
+
+[Modules]
+  [./FluidProperties]
+    [./simple_fluid]
+      type = SimpleFluidProperties
+      bulk_modulus = 1
+      density0 = 10
+      thermal_expansion = 0
+      viscosity = 11
+    [../]
+  [../]
+[]
+
+[Materials]
+  [./ppss]
+    type = PorousFlow1PhaseP
+    porepressure = pp
+    capillary_pressure = pc
+  [../]
+  [./massfrac]
+    type = PorousFlowMassFraction
+  [../]
+  [./simple_fluid]
+    type = PorousFlowSingleComponentFluid
+    fp = simple_fluid
+    phase = 0
+  [../]
+  [./porosity]
+    type = PorousFlowPorosityConst
+    porosity = 0.125
+  [../]
+
+  [./temperature]
+    type = PorousFlowTemperature
+    temperature = temp
+  [../]
+
+  [./temperature_nodal]
+    type = PorousFlowTemperature
+    at_nodes = true
+    temperature = temp
+  [../]
+[]
+
+[PorousFlow]
+  [./BCs]
+    [./left]
+      type = PorousFlowSourceBC
+      boundary = left
+      pressure = pp
+      T_in = 300
+      fp = simple_fluid
+      flux_function = -1
+    [../]
+  [../]
+[]
+
+[Preconditioning]
+  [./andy]
+    type = SMP
+    full = true
+    petsc_options_iname = '-ksp_type -pc_type -sub_pc_type -snes_max_it -sub_pc_factor_shift_type -pc_asm_overlap'
+    petsc_options_value = 'gmres asm lu 10000 NONZERO 2'
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = Newton
+  dt = 0.25
+  end_time = 1
+  nl_rel_tol = 1E-12
+  nl_abs_tol = 1E-12
+[]
+
+[Outputs]
+  file_base = s10
+  [./exodus]
+    type = Exodus
+    execute_on = 'initial final'
+  [../]
+[]
```

Comment:
OK, i just commented on this test, but now i just realised you're probably going to tidy this up anyway, and add more tests (and include in the "tests" file) and doco.  Oh well....

Reply:
yeah, still need some help with tests. i can do the jacobian one with no problems i think. but, i would probably need to redesign this one so it actually checks that the BC works and fails if somebody messes with the code. if you have ideas, i am all one ear...